src/PatchOrchestrationApplication/CoordinatorService/src/RepairManagerHelper.cs

```diff
@@ -274,6 +279,214 @@ internal async Task PrintRepairTasks(CancellationToken cancellationToken)
+            }
+        }
+
+        /// <summary>
+        /// Utility to find out ConsiderWarningAsError in Cluster manifest.
+        /// </summary>
+        private async Task<bool> ConsiderWarningAsErrorIsTrue()
+        {
+            string manifestString  = await this.fabricClient.ClusterManager.GetClusterManifestAsync();
+            XmlDocument clusterManifest = new XmlDocument();
+            string val = GetParamValueFromSection(clusterManifest, "HealthManager/ClusterHealthPolicy", "ConsiderWarningAsError");
+            bool flag;
+            if(Boolean.TryParse(val,out flag))
+            {
+                return flag;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Utility to parse cluster manifest.
+        /// </summary>
+        /// <sectionName>Name of the sectionName to search.</sectionName>
+        /// <parameterName>Name of the parameter to search.</parameterName>
+        /// <return>the value of parameter</return>
+        public string GetParamValueFromSection(XmlDocument doc, string sectionName, string parameterName)
+        {
+            XmlNode sectionNode = doc.DocumentElement?.SelectSingleNode("//*[local-name()='Section' and @Name='" + sectionName + "']");
+            XmlNode parameterNode = sectionNode?.SelectSingleNode("//*[local-name()='Parameter' and @Name='" + parameterName + "']");
+            XmlAttribute attr = parameterNode?.Attributes?["Value"];
+            return attr?.Value;
+        }
+
+        /// <summary>
+        /// Used to clear the events on Coordinator Service for nodes which are deleted from cluster.
+        /// </summary>
+        public async Task ClearOrphanEvents(CancellationToken cancellationToken)
+        {
+            try
+            {
+                ServiceHealth health = await this.fabricClient.HealthManager.GetServiceHealthAsync(this.context.ServiceName);
+                List<HealthEvent> healthEventsToCheck = new List<HealthEvent>();
+                foreach (var e in health.HealthEvents)
+                {
+                    if (e.HealthInformation.Property.Contains(WUOperationStatusUpdate))
+                    {
+                        healthEventsToCheck.Add(e);
+                    }
+                }
+                cancellationToken.ThrowIfCancellationRequested();
+
+                NodeList nodeList = await this.fabricClient.QueryManager.GetNodeListAsync(null, null, this.DefaultTimeoutForOperation, cancellationToken);
+                List<string> orphanProperties = new List<string>();
+                Dictionary<string, bool> propertyDict = new Dictionary<string, bool>();
+                if (healthEventsToCheck.Count == nodeList.Count)
+                {
+                    return;
+                }
```

Comment:
This may not be correct

Reply:
yes, but there will always be mismatch when nodes are added or deleted.

Reply:
I think we should remove this optimization.

---
In reply to: [281144758](https://github.com/microsoft/Service-Fabric-POA/pull/14#discussion_r281144758) [](ancestors = 281144758)